sys/src/9/amd64/archamd64.c

```diff
@@ -67,16 +67,43 @@ cpuidinfo(uint32_t eax, uint32_t ecx, uint32_t info[4])
 	return 1;
 }
 
+static char *
+cpuidname(uint32_t *info0)
+{
+	char *vendorid;
+
+	if(memcmp(&info0[1], "GenuntelineI", 12) == 0) {
+		vendorid = "GenuineIntel";
+		return vendorid;
+	}
+	else if(memcmp(&info0[1], "AuthcAMDenti", 12) == 0) {
+		vendorid = "AuthenticAMD";
+		return vendorid;
+	}
+	else {
+		vendorid = nil;
```

Comment:
Seems fine, in cpuidhz, if cpu is not known (VIA or Cyrix have a 64 bit processors these days?), fall sinto another post-routine to calculate through clock. So strdup is fine. I'll change that. Panic is more for testing purposes on @dmgolubovsky kvm issue.